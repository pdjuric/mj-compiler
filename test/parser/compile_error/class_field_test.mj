program main_test

class A {
    int a;

    {
        void b() {}
    }
}

class C extends A{
    char int;
    C c;
    char C;
    char TVF;
    char this;


    // ime se poklapa sa poljem ili metodom natklase
    char a;     // nedozvoljeno, u javi dozvljeno
    char b;     // dozvoljeno, ali problem u runtime?

}

{
    void main() {}
}