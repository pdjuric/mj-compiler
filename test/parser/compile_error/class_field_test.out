ERROR 14:26:25,622 - line  15 - Identifier TVF already defined in this scope.
ERROR 14:26:25,622 - line  16 - "this" is not an allowed variable name.
ERROR 14:26:25,622 - line  20 - Identifier a already defined in this scope.